File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/daos/IPlayerDao.java
package com.dkellycollins.triggerfinger.data.daos;

import com.dkellycollins.triggerfinger.data.entity.IPlayer;

public interface IPlayerDao {

    int create(int collidableId, int invincibleTimerId, int weaponId, int score, byte health);

    Iterable<IPlayer> retrieve();
    IPlayer retrieve(int id);
    IPlayer playerOne();

    void update(int id, int weaponId, int score, byte health);

    void delete(int id);
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/daos/IPlayerDao.kt
package com.dkellycollins.triggerfinger.data.daos

import com.dkellycollins.triggerfinger.data.entity.IPlayer

interface IPlayerDao : IActivityDao {

    fun create(collidableId: Int, invincibleTimerId: Int, weaponId: Int, score: Int, health: Byte): Int

    fun retrieve(): Iterable<IPlayer>
    fun retrieve(id: Int): IPlayer
    fun playerOne(): IPlayer

    fun update(id: Int, weaponId: Int, score: Int, health: Byte)

    fun delete(id: Int)
}